Sources/MessagingPush/PushClickHandler.swift

```diff
@@ -170,6 +180,25 @@ extension PushClickHandlerImpl {
         // continue swizzle
         cio_swizzle_didReceive(center, didReceive: response, withCompletionHandler: completionHandler)
     }
+
+    // Swizzled method that gets called when a push notification gets clicked or swiped away
+    @objc dynamic func cio_swizzle_willPresent(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        guard let _ = CustomerIOParsedPushPayload.parse(notificationContent: notification.request.content, jsonAdapter: jsonAdapter) else {
+            // push not sent from CIO. exit early and ignore request
+            return
+        }
+
+        if messagingPushConfig.showPushAppInForeground {
+            if #available(iOS 14.0, *) {
+                completionHandler([.list, .banner, .badge, .sound])
+            } else {
+                completionHandler([.badge, .sound])
+            }
+        }
+
+        // continue swizzle
+        cio_swizzle_willPresent(center, willPresent: notification, withCompletionHandler: completionHandler)
```

Comment:
Does this method call itself? I do not see any other occurrence of this method in the changed files. Am I missing out something ?

Reply:
See https://github.com/customerio/customerio-ios/pull/403#discussion_r1394524344